RecoMuon/MuonIsolation/plugins/ExtractorFromDeposits.h

```diff
@@ -26,6 +26,7 @@ class ExtractorFromDeposits : public reco::isodeposit::IsoDepositExtractor {
       const reco::TrackCollection & tracks) override;
   reco::IsoDeposit deposit (const edm::Event & ev, const edm::EventSetup & evSetup,
       const reco::Track & track) const override;
+  using reco::isodeposit::IsoDepositExtractor::deposit;
   virtual reco::IsoDeposit deposit (const edm::Event & ev, const edm::EventSetup & evSetup,
       const reco::TrackRef & track) const;
```

Comment:
I think that the type of the argument should be changed here to match the base method

Reply:
The base class does not handle this case. I think this is why this definition of deposit was added. The using statement tells the compiler this was intentional.

Reply:
I think that in this case the compiler correctly identified a problem with the code.
The ```   virtual reco::IsoDeposit deposit (const edm::Event & ev, const edm::EventSetup & evSetup, const reco::TrackRef & track) const ``` should instead use TrackBaseRef

Reply:
It is doing something https://github.com/gartung/cmssw/blob/e537dba5b7a90765339e78578f42404c2c4627f9/RecoMuon/MuonIsolation/plugins/ExtractorFromDeposits.cc#L49

Reply:
DXR indicates that deposit is never called.

Reply:
> It is doing something

yes, but my point is that it should overload the base class method and not create a new one.